src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -120,22 +131,24 @@ void FileIOMicroReadBenchmarkFixture::read_non_atomic_multi_threaded(benchmark::
     read_data.resize(NUMBER_OF_ELEMENTS);
     auto* read_data_start = std::data(read_data);
 
-    state.ResumeTiming();
-
     for (auto index = size_t{0}; index < thread_count; ++index) {
       auto from = batch_size * index;
       auto to = from + batch_size;
       if (to >= NUMBER_OF_ELEMENTS) {
         to = NUMBER_OF_ELEMENTS;
       }
-      threads[index] = (std::thread(read_data_using_read, from, to, filedescriptors[index], read_data_start));
+      threads[index] = std::thread(read_data_using_read, from, to, filedescriptors[index], read_data_start, std::ref(threads_ready_to_executed));
```

Comment:
now we don't measure the work distribution time which I would argue is something we should measure when comparing I/O for the same I/O size.
Perhaps it would be possible to create the threads without any work assigned to them in the non-timing preparation block. And then let them start as soon as they get assigned work?

Something like 

//create threads without assigned work
state.ResumeTiming();
for (auto index = size_t{0}; index < thread_count; ++index) {
       auto from = batch_size * index;
       auto to = from + batch_size;
       if (to >= NUMBER_OF_ELEMENTS) {
         to = NUMBER_OF_ELEMENTS;
       }
       // set fitting threads from and to values via a reference and let it start working

Reply:
I think that would also model the thread-pool approach one would probably take with assigning idle threads work whenever possible better.

Reply:
Lets talk about this tomorrow or even discuss it in the weekly. What would speak against simply timing the creation and then stopping the timer until we set the flag to execute it? Maybe we would measure then one iteration of the while loop

Reply:
What do you mean by “work distribution”?

ist that much more than passing two ints and a reference?

Reply:
>What do you mean by “work distribution”?
  ist that much more than passing two ints and a reference?

Jep, you're right. I was not considering the actual impact of those operations but rather the theoretical idea of measuring everything that belongs to I/O in this case. But I now agree that it's fine like this. 👍